src/mms_client/services/base.py

```diff
@@ -374,10 +376,11 @@ def request_many(
         Returns:    The multi-response from the MMS server.
         """
         # First, create the MMS request from the payload and data.
+        data_type = get_args(type(payload))[0] if isinstance(payload, list) else type(payload)
```

Comment:
So we ended up just getting the type of the first element. Makes sense if its always going to be a list of the same thing.

Reply:
Not quite. `typing.get_args` gets all the inner type arguments of a type (e.g. `get_args(List[int]` returns `(int,)`). So this is indexing into that tuple.